### PR TITLE
Encounter with flags

### PR DESCRIPTION
**Pull request by battmeech**
# Description

Added encounter generation along with flag parsing as this is quite a complex command with multiple parameters.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit test added for flag parsing

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -62,18 +62,18 @@ client.on('message', async (message: Message) => {
...
 
     if (message.content.indexOf(config.prefix) !== 0) {
         // On messages without prefix run these commands
         // Randomly meow when a message is received
-        if (Math.random() < guildSettings.randomSpeechProbability) {
-            speak(message)
-        }
+        // if (Math.random() < guildSettings.randomSpeechProbability) {
+        //     speak(message)
+        // }
```
Comment (djaustin) [inline src/index.ts:76]: Bring it back!

File: src/models/encounter.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Monster } from './monster'
+
+export class Encounter {
```
Comment (djaustin) [inline src/models/encounter.ts:3]: I think we should get adjusted XP in here as mentioned on Discord. From the 2 mins I spent looking at the DMG it should just be totalXP multiplied by some set value based on the monster count

File: src/models/monster.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    xp: number
+    environments: Environment[]
+}
+
+export type Environment = 'MOUNTAIN' | 'TREE'
+
+export const Monsters: Monster[] = [
+    { name: 'Goblin', environments: ['MOUNTAIN'], xp: 25 },
+    { name: 'Frost Giant', environments: ['MOUNTAIN'], xp: 2900 },
+]
```
Comment (djaustin) [inline src/models/monster.ts:12]: I wonder how we get the full set here. There's the [SRD ](https://dnd.wizards.com/articles/features/systems-reference-document-srd) which I think is free to use but I don't know if its accessible as an API.

**Review by djaustin — CHANGES_REQUESTED** [review]

File: src/commands/dies.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { Message } from 'discord.js'
 import moment from 'moment'
 import { findOrMakeGuild } from '../utils'
 
-export default async function die(message: Message) {
```
[the remark this answers is not in the document]
Comment (battmeech) [reply src/commands/dies.ts:5]: majority were named exports, thought I would go with majority here

File: src/index.ts (hunk 1 above)
Comment (battmeech) [reply src/index.ts:76]: Ah I deleted a lot of guild stuff on my mac so I didn't need mongo 👀

**Pull request closed without merging**